File: GitToolBox/src/main/java/zielu/gittoolbox/ui/statusBar/RepositoryActions.java
package zielu.gittoolbox.ui.statusBar;

import com.intellij.openapi.actionSystem.ActionGroup;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import git4idea.repo.GitRepository;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import zielu.gittoolbox.util.GtUtil;

public class RepositoryActions extends ActionGroup {
    private final GitRepository myRepository;

    public RepositoryActions(GitRepository repository) {
        super(GtUtil.name(repository), true);
        myRepository = repository;
    }

    @NotNull
    @Override
    public AnAction[] getChildren(@Nullable AnActionEvent e) {
        return new AnAction[] {
            new FetchAction(myRepository)
        };
    }
}

File: src/main/kotlin/zielu/gittoolbox/ui/statusbar/RepositoryActions.kt
package zielu.gittoolbox.ui.statusbar

import com.intellij.openapi.actionSystem.ActionGroup
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import git4idea.repo.GitRepository
import zielu.gittoolbox.util.GtUtil

internal class RepositoryActions(
  private val repository: GitRepository
) : ActionGroup(GtUtil.name(repository), true) {

  override fun getChildren(e: AnActionEvent?): Array<AnAction> {
    return StatusBarActions.actionsFor(repository).toTypedArray()
  }
}
